test/qi/function.cpp

```diff
@@ -0,0 +1,211 @@
+/*=============================================================================
+Copyright (c) 2016  Frank Hein, maxence business consulting gmbh
+
+Distributed under the Boost Software License, Version 1.0. (See accompanying
+file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+=============================================================================*/
+
+#include <boost/detail/lightweight_test.hpp>
+#include <boost/spirit/include/qi_operator.hpp>
+#include <boost/spirit/include/qi_char.hpp>
+#include <boost/spirit/include/qi_grammar.hpp>
+#include <boost/spirit/include/qi_rule.hpp>
+#include <boost/spirit/include/qi_string.hpp>
+#include <boost/spirit/include/qi_directive.hpp>
+
+#include <string>
+#include <iostream>
+#include <vector>
+#include "test.hpp"
+
+///////////////////////////////////////////////////////////////////////////
+// Implementation of a ternary conditional operator if_ for test and
+// demonstration purposes.
+//
+// The implementation shows how to implement an operator
+// which lives in a user's namespace (here mxc::qitoo).
+///////////////////////////////////////////////////////////////////////////
+
+namespace mxc { namespace qitoo {
```

Comment:
It is a reference to https://github.com/mxc-commons/MxcQitoo, where I publish qi extensions.